### ISSUE TITLE
demultiplex fastq files for individual cells from the single cell RNA seq data

### ISSUE THREAD

**Issue by lilycanfly** [Teichlab/tracer#58]
Hi,
I am working on the single cell RNA seq data from the 10x genomics pipeline. The fastq file generate by this pipeline put all the cells from one sample together. Is there any way I can demultiplex the fastq file for individual cells? Thanks a lot.

Lili

**Comment by mstubb** [comment]
Dear Lili,

Unfortunately TraCeR is not intended for use with data from the 10x genomics instrument. TCR reconstruction requires sequencing reads that cover the entire length of a transcript (eg from techniques such as SmartSeq2) rather than just the 3’ end. 

Best wishes,

Mike

**Comment by lilycanfly** [comment]
Dear Mike,

OK, Thanks a lot.

All the best,
Lili

**Comment by paolo-kunderfranco** [comment]
This means that results are unreliable? I was able to detect significant variants with 10x technology.

**Comment by mstubb** [comment]
It means that we didn't design TraCeR to work with 10x data, nor have we tested or validated it. 

I'm interested to hear more though. @paolo-kunderfranco please could you elaborate on what you've detected and which 10x assay type you're using. Is it the 3' or the 5' assay? Are you specifically enriching for VDJ sequences?

**Comment by paolo-kunderfranco** [comment]
we are using:
The Chromium™ Single Cell 3’ Solution
We are not enriching for VDJ sequences.
I extracted only cells belonging to a specific cluster, (T Cells), extracted only R2 sequences matching cells ID in R1 sequences, and gone through Tracer pipeline, with default kallisto, salmone, tracer settings..
Only a small subset of reads mapped, but at the end I got significant filtered TCRs sequences, especially for TCRa subunit ( and these are also expected in my T [cells):](url
[filtered_TCRs_AP01.txt](https://github.com/Teichlab/tracer/files/2235767/filtered_TCRs_AP01.txt)
)

I also tried to reverse complement all my R2 reads, results are exactly the same, except for TPM counts, any suggestion why?

best Paolo

**Comment by mstubb** [comment]
Did you demultiplex the reads and then put each cell separately through tracer?

**Comment by paolo-kunderfranco** [comment]
I demultiplexed all reads of cells belonging to a specific cell cluster, I run Tracer not at a single cell level, but at a cluster level, more or less 300 cells

**Comment by mstubb** [comment]
TraCeR is build on an assumption that it is running on sequences derived from a single cell. If you're violating that assumption and using reads from more than one cell it's possible that TraCeR will assemble artifactual sequences that aren't actually present - we haven't performed any tests or validation on how it might behave when intentionally run on sequences from multiple cells. Also, the `filtered` results are likely to be meaningless since one would expect more than two sequences for each locus due to diversity in the cluster. 

I suppose that if the T cells in a cluster are 100% clonally related then TraCeR *should* reconstruct the correct sequences (although, again, we haven't tested this) but that's a very specific biological situation. 

In short, I don't recommend this use of TraCeR unless you're very confident that it makes sense in your particular biological situation and, even then, you should validate any important findings with a different method.
